src/presentation/controllers/device-controller.ts

```diff
@@ -94,4 +94,14 @@ export class DeviceController extends BaseController {
       this.httpErrorHandler.handleError(response, error as Exception)
     }
   }
+
+  @DeleteRequest('/:deviceId')
+  public async deleteDevice(request: Request, response: Response): Promise<void> {
+    try {
+      const {deviceId} = request.params
```

Comment:
Would be nice with a syntax similar to `const deviceId: string = request.params.deviceId`.

Reply:
To address your comment about type safety I did as you suggested and moved from destructuring assignment to type assertion